[Ryan Guai] iP

ryanguai opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)
*****
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ryanguai committed:
Greet, Echo, Exit

ryanguai committed:
Add, List

ryanguai committed:
Added Task class

ryanguai committed:
Added markAsDone, markNotDone functions

ryanguai committed:
Completed deadline task

ryanguai committed:
Added Event, Todo

ryanguai committed:
Filled in input and expected output for automated testing

ryanguai committed:
Handle errors

ryanguai committed:
Delete

ryanguai committed:
Remove redundant file

ryanguai committed:
Create and write to text file when tasks are added

ryanguai committed:
Save complete

ryanguai committed:
Add support for Dates and Times

ryanguai committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

ryanguai committed:
Refactor code to extract closely related code as classes

ryanguai committed:
Reorganize into packages

ryanguai committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

ryanguai committed:
Add JUnit tests

ryanguai committed:
Support packaging as JAR

ryanguai committed:
Add documentation

ryanguai committed:
Follow coding standard

ryanguai committed:
Add ability to find task by searching for keyword

ryanguai committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/parser/Parser.java

ryanguai committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/duke/task/Event.java

ryanguai committed:
Use JavaFX to add a GUI for Duke

ryanguai committed:
Removed I/O automated testing and functions used for CLI

maryjess commented on src/main/java/duke/command/TaskList.java:
```diff
@@ -0,0 +1,95 @@
+        taskList.add(task);
+    }
+
+    public ArrayList<Task> searchMatches(String keyword) {
```
Perhaps method name of `searchTasks` or `searchMatchingKeyword` is better?

maryjess commented on src/main/java/duke/command/TaskList.java:
```diff
@@ -0,0 +1,95 @@
+
+    public ArrayList<Task> searchMatches(String keyword) {
+        ArrayList<Task> matchingList = new ArrayList<>();
+        for (Task task: taskList) {
```
Should we add a spacebar here, after `task`?

maryjess commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,124 @@
+        String dukeResponse = "";
+            try {
+                switch (response) {
+                    case "find": {
```
Perhaps this line should have one less indent?
https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Note%20there%20is%20no%20indentation%20for%20case%20clauses.

maryjess commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+            FileWriter myWriter = new FileWriter(this.filePath);
+            StringBuilder output = new StringBuilder();
+            ArrayList<Task> tasks = taskList.getTasks();
+            for (Task task:tasks) {
```
Perhaps can add spacebar here as well?

maryjess commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Perhaps the instance variable name can be more descriptive, e.g. `deadlineDate` or `date`?

maryjess commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,67 @@
+
+    /**
+     * Marks task as done and prints message.
+     */
```
Perhaps can add `@return` here, as part of the Javadoc?

maryjess reviewed (commented):
Great work on your code so far! 👍

ryanguai committed:
Update mainClassName in gradle build file

The previous mainClassName was incorrectly used as the default example given in the tutorial.
Let's change it so that the JAR file generated is correct.

ryanguai committed:
Use assertions

Let's enable assertions to check that important assumptions hold
throughout the code.

ryanguai committed:
Improve code quality

Let's refactor code to remove unnecessary import statements,
avoid magic literals and complicated expressions.

ryanguai committed:
Merge pull request #1 from ryanguai/branch-A-Assertions

Use assertions

ryanguai committed:
Merge branch 'branch-A-CodeQuality'

ryanguai committed:
Prioritizing items

Let's provide a way to attach priorities to items. When the task list is printed, it would be
useful to sort by priority.

ryanguai committed:
Make minor changes to improve code quality, add user guide and product screenshot

ryanguai committed:
Correct typos in UG

ryanguai committed:
Fix bug for deadline tasks, change name of JAR file

ryanguai committed:
Fix bug with saving to and loading from save file

ryanguai committed:
Perform final fixes based on bugs detected through smoke testing
